huaweicloud/services/cci/data_source_huaweicloud_cci_namespaces.go

```diff
@@ -0,0 +1,254 @@
+	if err != nil {
+		return nil, err
+	}
+
+	result := make([]map[string]interface{}, len(resp))
+	for i, network := range resp {
+		result[i] = map[string]interface{}{
+			"id":                network.Metadata.Id,
+			"name":              network.Metadata.Name,
+			"vpc_id":            network.Spec.AttachedVPC,
+			"security_group_id": network.Metadata.Annotations["network.alpha.kubernetes.io/default-security-group"],
+			"subnet_id":         network.Spec.SubnetID,
+			"subnet_network_id": network.Spec.NetworkID,
+			"subnet_cidr":       network.Spec.Cidr,
+		}
+	}
+	return result, nil
+}
+
+func getCciNamespaceIds(nsList []namespaces.Namespace) []string {
+	result := make([]string, len(nsList))
+	for i, ns := range nsList {
+		result[i] = ns.Metadata.UID
+	}
+	return result
+}
+
+func setDataCciNamespacesParams(d *schema.ResourceData, c *golangsdk.ServiceClient, nsList []namespaces.Namespace) error {
+	result := make([]map[string]interface{}, 0, len(nsList))
+	for _, ns := range nsList {
+		if isNamespaceParamMatch(d, ns) {
+			nsParams := map[string]interface{}{
+				"id":                        ns.Metadata.UID,
+				"name":                      ns.Metadata.Name,
+				"type":                      ns.Metadata.Annotations.Flavor,
+				"enterprise_project_id":     ns.Metadata.Labels.EnterpriseProjectID,
+				"auto_expend_enabled":       ns.Metadata.Annotations.AutoExpend,
+				"warmup_pool_size":          ns.Metadata.Annotations.PoolSize,
+				"recycling_interval":        ns.Metadata.Annotations.RecyclingInterval,
+				"container_network_enabled": isContainNetworkEnabled(ns.Metadata.Annotations.NetworkEnable),
+				"rbac_enabled":              ns.Metadata.Labels.RbacEnable,
+				"created_at":                ns.Metadata.CreationTimestamp,
+				"status":                    ns.Status.Phase,
+			}
+
+			netResp, err := getCciNetworks(c, ns.Metadata.Name)
+			if err != nil {
+				return err
+			}
+			nsParams["network"] = netResp
+
+			result = append(result, nsParams)
+		}
+	}
+
+	return d.Set("namespaces", result)
+}
+
+func dataSourceCciNamespacesRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	region := config.GetRegion(d)
+	client, err := config.CciV1Client(config.GetRegion(d))
+	if err != nil {
+		return fmtp.DiagErrorf("Error creating HuaweiCloud CCI v1 client: %s", err)
+	}
+	betaClient, err := config.CciV1BetaClient(config.GetRegion(d))
+	if err != nil {
+		return fmtp.DiagErrorf("Error creating HuaweiCloud CCI v1 beta1 client: %s", err)
+	}
+
+	pages, err := namespaces.List(client, namespaces.ListOpts{}).AllPages()
+	if err != nil {
+		return fmtp.DiagErrorf("Error finding the namespace list from the server: %s", err)
+	}
+	nsList, err := namespaces.ExtractNamespaces(pages)
+	if err != nil {
+		return fmtp.DiagErrorf("Error extracting HuaweiCloud CCI namespaces: %s", err)
+	}
+
+	ids := getCciNamespaceIds(nsList)
+	d.SetId(hashcode.Strings(ids))
```

Comment:
the ids is not filtered， is it ok to use the hashcode of ids as data_source ID?

Reply:
yeah, that's a wrong code.
fixed in `29f81a5`.